### ISSUE TITLE
Enhancement Request: Read and Write Contract Interactions on Verified Contracts

### ISSUE THREAD

**Issue by mshakeg** [hashgraph/hedera-mirror-node-explorer#782]
### Problem

The explorer lacks a critical functionality: the ability for users to directly interact with verified smart contracts through the explorer interface. This limitation hinders the ease and efficiency with which developers and users can engage with contracts, especially when compared to platforms like Etherscan which offer this feature. The absence of direct read and write interactions for verified contracts (where the ABI is known) on the explorer restricts the full potential of user engagement and contract accessibility.

### Solution

To address this, I propose an enhancement to the explorer that would enable read and write interactions on the contract page for verified smart contracts. This feature would be contingent on the contract being verified and its ABI being available. This enhancement would leverage the existing wallet connection capability of the explorer, allowing users to perform write operations on the smart contracts directly from the explorer interface.

**Comment by mshakeg** [comment]
Hey @svienot @ericleponner do you think this feature will be implemented soon. It's tedious having to interact with contracts on Hedera using tools such as this:
https://neptunemutual.com/web3-tools/solidity-abi-encoder-online/

**Comment by Nana-EC** [comment]
Hi @mshakeg 
Thanks for the issue and highlighting your current user flow.
Exposing the requested features to improve community experience has always been a plan for the smart contract verification and block explorer story.
Having this data point is great context for future planning.
Thanks

**Comment by mshakeg** [comment]
This should work for proxy contracts as well similar to how etherscan allows users to select `Read as Proxy` or `Write as Proxy` if it detects that the contract is a proxy contract e.g.

https://ftmscan.com/address/0xE02768682010984450a36f24eCcDb4A1Eb0ce869#readProxyContract

**Comment by mshakeg** [comment]
Hi @ericleponner I took a look at what's implemented and it's good, but I just wanted to suggest a few improvements to the current design(inspired by [etherscan](https://etherscan.io/address/0x1F98431c8aD98523631AE4a59f267346ea31F984#writeContract)):

1. have a read and write tab which separates view and non-view functions
2. number the functions
3. show the function selector
4. permalinks to a function

An idea unrelated to what etherscan implements(which might be a bit excessive) would be to also allow specifying the block number at which to make a static eth call to a view function instead of `latest`(which can remain the default behaviour)

**Comment by mshakeg** [comment]
Hi @ericleponner any idea when [this request](https://github.com/hashgraph/hedera-mirror-node-explorer/issues/782#issuecomment-1908955006) will be implemented? I want to be able to read&write as proxy, for example on the following `TransparentUpgradeableProxy` with uses the implementation below.

[TransparentUpgradeableProxy](https://hashscan.io/mainnet/contract/0.0.4544494)
[ProxyAdmin](https://hashscan.io/mainnet/contract/0.0.4544491)
[RebalancerFactory implementation](https://hashscan.io/mainnet/contract/0.0.4544461)

**Comment by ericleponner** [comment]
Hi @mshakeg 
I created  #964: let's continue discussion with this new issue.
